NDTensors/ext/examples/NDTensorAMDGPU.jl

```diff
@@ -0,0 +1,85 @@
+## Code adapted from NDTensors/ext/examples/NDTensorCUDA.jl
+
+using AMDGPU
+using NDTensors
+using ITensors
+using ITensors:
+  Index, ITensor, randomMPO, randomMPS, inner, orthogonalize, qr, siteinds, svd
+using Test: @test
+
+function main()
+  # using ITensorGPU
+  cpu = NDTensors.cpu
+  gpu = NDTensors.roc
+  # Here is an example of how to utilize NDTensors based tensors with AMDGPU datatypes
+  i = Index(2)
+  j = Index(5)
+  k = Index(3)
+  l = Index(6)
+
+  dim1 = (i, j, l)
+  dim2 = (j, k)
+
+  # Create 2 ITensors with AMDGPU backends
+  A = ITensor(randomTensor(ROCArray, dim1))
+  B = ITensor(randomTensor(ROCArray, dim2))
+
+  # Contract the two tensors
+  C = A * B
+  A = cpu(A)
+  B = cpu(B)
+  @test cpu(C) ≈ A * B
+  @test eltype(C) == Float64
+
+  # Create 2 ITensors on CPU with different eltypes
+  A = ITensor(Float32, dim1)
+  B = ITensor(Float64, dim2)
+
+  fill!(A, randn())
+  fill!(B, randn())
+
+  # Convert the ITensors to GPU
+  cA = gpu(A)
+  cB = gpu(B)
+
+  # Check that backend of contraction is GPU
+  @test A * A ≈ cpu(cA * cA)
+  @test B * B ≈ cpu(cB * cB)
+  @test A * B ≈ cpu(cA * cB)
+  @test B * A ≈ cpu(cB * cA)
+
+  dim3 = (l, k)
+  dim4 = (i,)
+  cC = ITensor(randomTensor(ROCArray{Float64,AMDGPU.Runtime.Mem.HIPBuffer}, dim3))
+  cD = ITensor(Tensor(ROCArray{Float32}, dim4))
+  fill!(cD, randn())
+
+  # Its possible to compute QR of GPU tensor
+  cq = qr(cA, (i,), (j, l))
+  A ≈ cpu(cq[1]) * cpu(cq[2])
+
+  res = ITensors.svd(A, (i,), (j, l))
+  @show res
+
+  s = siteinds("S=1/2", 8)
+  m = randomMPS(s; linkdims=4)
+  cm = gpu(m)
+
+  @test inner(cm', cm) ≈ inner(m', m)
+
+  H = randomMPO(s)
+  cH = gpu(H)
+  @test inner(cm', cH, cm) ≈ inner(m', H, m)
+
+  m = orthogonalize(m, 1)
+  cm = gpu(orthogonalize(cm, 1))
+  @test inner(m', m) ≈ inner(cm', cm)
+
+  H = orthogonalize(H, 1)
+  cH = gpu(cH)
+
+  @test inner(cm', cH, cm) ≈ inner(m', H, m)
```

Comment:
I would prefer to remove these tests that rely on MPO/MPS functionality. We are planning to disentangle code based on MPO/MPS types from the rest of the ITensors.jl codebase and move it into a separate repository, and having tests that intermingle basic ITensor functionality from MPO/MPS functionality will make that more challenging.

Reply:
Oh I see these are examples, not tests.

Anyway, I'm not really a fan of having these extension-specific example files, let's just remove this file for now.

Reply:
Note for myself. I haven't looked at the example file yet.

Reply:
@mtfishman I'm deleting this now, thanks I wasn't sure what to do with this file.